### PR TITLE
Feature/support tvos

### PR DESCRIPTION
**Pull request by bimawa**
Related with grpc/grpc#14173

Pls, check these changes.

### CODE REVIEW

**Comment by enricmacias** [comment]
Hey guys! There is any chance this branch is merged to master anytime soon?
Thanks!

**Comment by MrMage** [comment]
This can only be merged once the vendored versions of gRPC and BoringSSL have also been updated by running https://github.com/grpc/grpc-swift/blob/master/vendor-grpc.sh and https://github.com/grpc/grpc-swift/blob/master/vendor-boringssl.sh, which has not been done yet.

**Comment by bimawa** [comment]
@MrMage Hi, pls check my PR.

**Comment by MrMage** [comment]
@bimawa looks like this fails to compile: https://travis-ci.org/grpc/grpc-swift/jobs/510133495.

I think @rebello95 had tried to upgrade our vendored gRPC version recently, but encountered issues as well. Michael, could you comment what your last status with that was?

**Comment by bimawa** [comment]
@MrMage yes, I need help with it. Maybe I can compile it with some command on the local machine?

**Comment by rebello95** [comment]
> I think @rebello95 had tried to upgrade our vendored gRPC version recently, but encountered issues as well. Michael, could you comment what your last status with that was?

I added a script [here](https://github.com/grpc/grpc-swift/blob/master/scripts/vendor-all.sh) that should help with updating the vendored dependencies. I started looking into updating them manually, but ran into some issues getting BoringSSL to compile properly (and haven't had a chance to look at it since). 

Quite a few changes went in to BoringSSL upstream (like [this one](https://github.com/grpc/grpc/commit/b24b212ee585d376c618235905757b2445ac6461)), so it will likely require some manual changes.

**Comment by bimawa** [comment]
@rebello95 @MrMage Guys I use that script for update vendors. And second moment, with @muxi we update our BoringSSL, official boringSSL not support tvos. Maybe we don't have to grab this dependency, its included in grcp-core?
I tested it with that podspec in my PR, its work fine.

**Comment by rebello95** [comment]
>  with @muxi we update our BoringSSL, official boringSSL not support tvos

Can you rephrase this? Not sure I understand what you mean.

> Maybe we don't have to grab this dependency, its included in grcp-core? I tested it with that podspec in my PR, its work fine.

Vendored dependencies are used by SwiftPM/Carthage, not CocoaPods (and CI uses SwiftPM). You can test locally using `make project` and building `SwiftGRPC.xcodeproj` targets. Additionally, you can take a look at the Travis CI failures by clicking `Details` next to the job(s) on this PR for logs and details on the failures. Looks like openssl (BoringSSL) is causing compilation failures.

**Comment by bimawa** [comment]
>Can you rephrase this? Not sure I understand what you mean.

We have a fork of BoringSSL, and its called now as BoringSSL-GRPC in grpcCore. Which support tvos and other features. So BoringSSL, which included in vendor script will not support swift-grpc for tvos. 

>Vendored dependencies are used by SwiftPM/Carthage

Ok, so we have to take BoringSSL-GRPC from grpc repo, not BoringSSL from the official. IMHO.

**Comment by rebello95** [comment]
Using the `BoringSSL-GRPC` from the `grpc` repo sounds reasonable to me, especially since that's what CocoaPods seems to be using upstream. Would you mind making this change and updating the script(s) for future use?

**Comment by bimawa** [comment]
Yes, but I need time for research that scripts.

**Comment by timburks** [comment]
I'm looking into the vendoring problems, hoping to have an update by tomorrow at the latest.

**Comment by bimawa** [comment]
@timburks It's will be cool! Thank you!

**Comment by timburks** [comment]
#417 updates the vendored gRPC to 1.19.1 (with tests currently passing on Mac and failing on Linux). It doesn't use the BoringSSL-GRPC from the gRPC repo, but hopefully makes that easier. As far as I can tell, BoringSSL-GRPC is completely defined by  [grpc/src/objective-c/BoringSSL-GRPC.podspec](https://github.com/grpc/grpc/blob/master/src/objective-c/BoringSSL-GRPC.podspec), which contains over 4000 lines of inline source files. I think that will make vendoring updates difficult. @muxi could those files be separated out of the podspec?

**Comment by muxi** [comment]
@bimawa - thanks for the reminder.

@timburks - I am not sure if there is a way to separate the inline sources out. The reason for us using BoringSSL-GRPC is because users using both gRPC and OpenSSL are having conflict problem. We are using these podspec-inline sources to modify BoringSSL symbols to avoid symbol conflict. Since we do not have control over BoringSSL code base, this is the only place we know of that could affect the BoringSSL archive. If you know another way please let us know.

**Comment by MrMage** [comment]
@bimawa now that https://github.com/grpc/grpc-swift/pull/417 is merged, would you be interested in creating a new PR with only the changes necessary to support tvOS (if any)?

**Comment by bimawa** [comment]
@MrMage Can I update from upstream this PR?

**Comment by MrMage** [comment]
@bimawa if I remember correctly, the changes needed to support tvOS were quite small; most of the changes in this PR were the new gRPC version. I would prefer if you created a new PR with just the changes for tvOS.

**Comment by bimawa** [comment]
@MrMage ok.
